src/components/UserInfo/UserInfo.tsx

```diff
@@ -1,5 +1,24 @@
-import React from 'react';
+import { User } from '../../types/user';
+import { Comment } from '../../types/comment';
 
-export const UserInfo: React.FC = () => (
-  <>Put the user here</>
-);
+type Props = {
+  user: User | Comment,
+  dataType: 'post' | 'comment',
+};
```

Comment:
maybe here is the problem. You have only one User or null for Post and array of Comments or null per Post. So if you will tie Comments and User to Post from the start, you will have interface of Post, that already includes appropriate User and Comments

Reply:
commonly there is a sense about passing datas through "the forest" but not in this block.

Reply:
> maybe here is the problem. You have only one User or null for Post and array of Comments or null per Post. So if you will tie Comments and User to Post from the start, you will have interface of Post, that already includes appropriate User and Comments

commonly there is a sense about passing datas through "the forest" but not in this block.

Reply:
but why problem is here? if u check html structure of the code, it's totally ok
and in the api post has rly 5 comments, i cant. understand why test expects 3

Reply:
its not here)